elastica/modules/contact.py

```diff
@@ -0,0 +1,171 @@
+__doc__ = """
+Contact
+-------
+
+Provides the contact interface to apply contact forces between objects
+(rods, rigid bodies, surfaces).
+"""
+
+
+class Contact:
+    """
+    The Contact class is a module for applying contact between rod-like objects . To apply contact between rod-like objects,
+    the simulator class must be derived from the Contact class.
+
+        Attributes
+        ----------
+        _contacts: list
+            List of contact classes defined for rod-like objects.
+    """
+
+    def __init__(self):
+        self._contacts = []
+        super(Contact, self).__init__()
+        self._feature_group_synchronize.append(self._call_contacts)
+        self._feature_group_finalize.append(self._finalize_contact)
+
+    def detect_contact_between(self, first_system, second_system):
```

Comment:
type hints? since you already have in doc string

Reply:
ok